src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -216,6 +225,28 @@ jint ShenandoahHeap::initialize() {
                               "Cannot commit heap memory");
   }
 
+  BarrierSet::set_barrier_set(new ShenandoahBarrierSet(this, _heap_region));
```

Comment:
It is because `ShenandoahBarrierSet` now hosts `ShenandoahCardTable`. `ShenandoahCardTable` needs to know the size and location of the heap. Nothing will use the barrier set at this point.